### PR TITLE
added flag to set the image center as origin

### LINKED ISSUE #111: Add way to modify the image origin to its center
There is currently not way to have a nice looking fzf preview with ueberzug due not being able to center the image on either axis. The simplest fix is to add a way to modify the image's origin, meaning the position sent to ueberzug is the center of the displayed image. I'm sure fzf previews aren't the only thing that could use this feature.

For our case (lobster's rewrite), the simplest way would be to have a parameter like `--origin center` or something similar, so we don't have to get the image's size either.

### PR DESCRIPTION
**Pull request by ek234**
Closes #111 .
Added a flag which sets the origin to the center of the image. x, y is relative to the origin. If the flag is absent, the origin is set to the top left of the image.

Tested on kitty and alacritty. Might have missed some tests as idk the full functionality of ueberzugpp.

Let me know if you would like to change the variable names, types, formatting, etc. to fit the codebase.

### CODE REVIEW

File: include/dimensions.hpp
Comment (jstkdng) [inline include/dimensions.hpp:1]: there's no need to add the flag to dimensions, it should already be on the flags singleton each image class has.

File: src/main.cpp
Comment (jstkdng) [inline src/main.cpp:1]: hmm, shouldn't the flag be on the layer subcommand? otherwise only projects that use the unix socket can benefit from this

**Comment by diniamo** [comment]
He joined chomsky's ~~slavery operation~~  dev team as the cpp guy, so we told him he could take a shot at this.

**Comment by jstkdng** [comment]
thanks... I guess? Don't know who that is though

File: include/dimensions.hpp
Comment (ek234) [reply include/dimensions.hpp:1]: i felt it was directly related to the dimensions class as it modifies the value of x , y wrt orig_x, orig_y. So, i added it to the class. ideally, i would have set the exact value for x, y in the constructor of dimensions itself, however, dimensions doesnt not seem to have the exact width and height of the image (from what i understand) so i am adding the flag to the dimensions class and correcting the value of x, y at the Image class when width and height is known.

File: src/main.cpp
Comment (ek234) [reply src/main.cpp:1]: ahh i was expecting something like this. i dont know all use cases of uz so i missed this. how can i use uz other than unix sockets?
i will set the flag to the layer subcommand,

Comment (jstkdng) [reply src/main.cpp:1]: you can use `ueberzugpp` by sending commands through stdin.

**Comment by diniamo** [comment]
Oh whoops, that's justchokingaround. Sorry it wasn't appropriate.

**Comment by ek234** [comment]
@jstkdng made changes you suggested and tested on kitty and alacritty (X11 with opencv).

i need a clarification. why is 1 added to the x and y values of dimensions in the Sixel constructor (and similarly for chafa)? i am asking because i wanna know the intended behaviour - i am using using floor() but depending on the intended behaviour, this might have to be changed to ceil().

**Comment by jstkdng** [comment]
>why is 1 added to the x and y values of dimensions in the Sixel constructor

all the tests I've done resulted in some of the filemanager ui getting overwritten by the sixel content, adding 1 seems to fix it

**Comment by ek234** [comment]
aight. lemmino if this code passes your tests.

**Comment by jstkdng** [comment]
eh, I'm too lazy to test a feature that only one will be used by one project. I hope that it's in your best interest for it to work on all terminals that would be supported by the lobster rewrite. If another project comes by with a request similar to this one I might change the current implementation and test myself.

In the meantime, merging.

**Pull request merged**